syft/pkg/cataloger/config.go

```diff
@@ -0,0 +1,22 @@
+package cataloger
+
+import (
+	"github.com/anchore/syft/syft/pkg/cataloger/java"
+)
+
+type Config struct {
+	Search SearchConfig
+}
+
+func DefaultConfig() Config {
+	return Config{
+		Search: DefaultSearchConfig(),
+	}
+}
+
+func (c Config) Java() java.Config {
+	return java.Config{
+		SearchUnindexedArchives: c.Search.IncludeUnindexedArchives,
+		SearchIndexedArchives:   c.Search.IncludeIndexedArchives,
```

Comment:
although this PR is java specific, these would be common variables for #246 probably, would you say?

Reply:
That's what I'm thinking may be the case as well. 

I had the config item as `packages.java.search-unindex-archives` but changed it to `packages.search-unindex-archives` in case we add this functionality to other catalogers or full up include this as a configurable to #246 . Also doesn't hurt #246 if different configurables are needed.